src/lib/Tisch/Run.hs

```diff
@@ -128,13 +128,13 @@ data Perm
 -- Note that 'Conn' is not thread-safe, you are encouraged to maintain a
 -- multithreaded pool of 'Conn' instead. See "Data.Pool" from the @ex-pool@
 -- package.
-newtype Conn (perms :: [Perm]) = Conn Pg.Connection
+newtype Conn (perms :: [Perm]) d = Conn Pg.Connection
```

Comment:
Could you make `d` the first argument? The motivation for this is twofold: To match `Query`'s argument order thus reducing the surprise factor for users of this, and because partially applying `Conn d` is more likely to be useful than partially applying `Conn perms`. 

Also, could you add an explicit `:: k` to `d`? It makes things easier to reason about for readers of this code, seeing how kind and type heavy this library is.  
 
Also, could you change the documentation here so that it mentions this `d` argument, explains it purpose, and mention (as just one example) how in practice this `d` will be the same `d` that shows up in the `Query` type?

Thanks!